### ISSUE TITLE
How to Make Changes in Frontend Code/Layout OctoBot

### ISSUE THREAD

**Issue by mkamranmazhar** [Drakkar-Software/OctoBot#2583]
Hello,

I've successfully running OctoBot with Docker on my server.

I'm looking to make some changes to its frontend layout. However, I'm encountering difficulty understanding where the frontend files are located.

someone please guide me on how to locate the frontend files within the source code?

Thank you.

**Comment by Herklos** [comment]
Hi @mkamranmazhar,

You can find the source code of the OctoBot web interface here https://github.com/Drakkar-Software/OctoBot-Tentacles/tree/master/Services/Interfaces/web_interface.

These files are automatically installed at first startup as modules named "tentacles"

**Comment by mkamranmazhar** [comment]
> These files are automatically installed at first startup as modules named "tentacles"

Thank you @Herklos 

These files will also be automatically installed if I run this via Docker, right?

**Comment by Herklos** [comment]
Right, you can find them at `/octobot/tentacles` inside your docker container

**Comment by mkamranmazhar** [comment]
@Herklos,

Please check image files are not available in this folder.

<img width="911" alt="octobot-docker" src="https://github.com/Drakkar-Software/OctoBot/assets/106512746/06ab376a-fcb4-49dd-86b9-3c61bacdc12f">

Thank you.

**Comment by Herklos** [comment]
> @Herklos,
> 
> Please check image files are not available in this folder.
> 
> <img alt="octobot-docker" width="911" src="https://private-user-images.githubusercontent.com/106512746/320131119-06ab376a-fcb4-49dd-86b9-3c61bacdc12f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIzNDkzNzYsIm5iZiI6MTcxMjM0OTA3NiwicGF0aCI6Ii8xMDY1MTI3NDYvMzIwMTMxMTE5LTA2YWIzNzZhLWZjYjQtNDlkZC04NmI5LTNjNjFiYWNkYzEyZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDA1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQwNVQyMDMxMTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01M2I0Nzc5ZWE0NTU5M2E4ODAwZDRhMTMyMWYxNDJiYjM5NTgxNDcwNDZhMGU4NDFhOWFlZTNjZGU4M2ExNmY5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.J8dr1wJqRw9EvEZEgBnPbut22pqgd9rGnuI-fp9bR48">
> Thank you.

They are installed at first startup

**Comment by mkamranmazhar** [comment]
> They are installed at first startup

I've already done first startup, please check image & files are still not available in this location `/octobot/tentacles`.

both images attached.

<img width="1419" alt="octobot-first-run" src="https://github.com/Drakkar-Software/OctoBot/assets/106512746/4202be38-2f68-453e-af24-415330857aed">

<img width="914" alt="octobot-docker" src="https://github.com/Drakkar-Software/OctoBot/assets/106512746/6a4ad781-8ef1-494a-83bb-c08a5cc4c22a">

Thank you.

**Comment by Herklos** [comment]
The second screenshot contains the files available in the OctoBot docker image. It doesn't represent the files on your computer/server. A docker image once started can create new files and directories.

**Comment by mkamranmazhar** [comment]
@Herklos,

Indeed, you're correct. Once the Docker image is started, files are automatically generated within the Docker container. However, I'm curious about the location of the frontend code files within the Docker container. Obviously, frontend code files are also generated, which is why the frontend is displayed when I start it in the browser.

Thank you.

**Comment by Herklos** [comment]
They are packaged from https://github.com/Drakkar-Software/OctoBot-Tentacles/tree/master/Services/Interfaces/web_interface and then downloaded from our servers when a bot starts.

**Comment by mkamranmazhar** [comment]
Thank you for update, @Herklos 

> They are packaged before being downloaded from your servers to our server?

And if so, does this mean I need to build the Docker image myself if I want to make changes to the source code?

**Comment by Herklos** [comment]
No you can just mount a volume on /tentacles (like https://github.com/Drakkar-Software/OctoBot/blob/master/Dockerfile#L66) and modify the web interface source files
